pkg/util/testingjobs/statefulset/wrappers.go

```diff
@@ -96,8 +98,24 @@ func (ss *StatefulSetWrapper) Name(n string) *StatefulSetWrapper {
 	return ss
 }
 
-func (ss *StatefulSetWrapper) WithOwnerReference(ownerReference metav1.OwnerReference) *StatefulSetWrapper {
-	ss.OwnerReferences = append(ss.OwnerReferences, ownerReference)
+// UID updates the uid of the StatefulSet
+func (ss *StatefulSetWrapper) UID(uid string) *StatefulSetWrapper {
+	ss.ObjectMeta.UID = types.UID(uid)
+	return ss
+}
+
+// OwnerReference adds a ownerReference to the StatefulSet.
+func (ss *StatefulSetWrapper) OwnerReference(ownerName string, ownerGVK schema.GroupVersionKind) *StatefulSetWrapper {
+	ss.ObjectMeta.OwnerReferences = append(
```

Comment:
Can we reuse AppendOwnerReference in pkg/util/testing/wrappers.go?

Reply:
Done